pkg/vpcmodel/nodesExplainability.go

```diff
@@ -297,6 +315,91 @@ func computeActualRules(rulesLayer *rulesInLayers, filtersExternal map[string]bo
 	return &actualRules, directionEnabled
 }
 
+// computes combined list of rules, both deny and allow
+func (details *rulesAndConnDetails) computeCombinedActualRules() {
+	for _, singleSrcDstDetails := range *details {
+		actualRulesIngress := mergeAllowDeny(singleSrcDstDetails.actualAllowRules.ingressRules,
+			singleSrcDstDetails.actualDenyRules.ingressRules)
+		actualRulesEgress := mergeAllowDeny(singleSrcDstDetails.actualAllowRules.egressRules,
+			singleSrcDstDetails.actualDenyRules.egressRules)
+		actualRules := &rulesConnection{actualRulesIngress, actualRulesEgress}
+		singleSrcDstDetails.actualMergedRules = actualRules
+	}
+}
+
+// merges two rulesInLayers - for merging deny and allow for ingress and egress
+func mergeAllowDeny(allow, deny rulesInLayers) rulesInLayers {
+	allowDenyMerged := rulesInLayers{}
+	for _, layer := range filterLayers {
+		allowForLayer, ok1 := allow[layer]
+		denyForLayer, ok2 := deny[layer]
+		if !ok1 || !ok2 {
+			switch {
+			case ok1: // layer relevant only for deny
+				allowDenyMerged[layer] = allowForLayer
+			case ok2: // layer relevant only for allow
+				allowDenyMerged[layer] = denyForLayer
+			}
+			continue
+		}
+		mergedRulesInLayer := []RulesInFilter{} // both deny and allow in layer
+		// gets all indexes, both allow and deny, of a layer (e.g. indexes of nacls)
+		allIndexes := getAllIndexesForFilter(allowForLayer, denyForLayer)
+		for _, filterIndex := range allIndexes {
+			allowRules := getRulesInFilter(allowForLayer, filterIndex)
+			denyRules := getRulesInFilter(denyForLayer, filterIndex)
+			// only one of them can be nil if we got here
+			switch {
+			case denyRules == nil:
+				mergedRulesInLayer = append(mergedRulesInLayer, *allowRules)
+			case allowRules == nil:
+				mergedRulesInLayer = append(mergedRulesInLayer, *denyRules)
+			default: // none nil, merge
+				mergedRules := []int{}
+				mergedRules = append(mergedRules, allowRules.Rules...)
+				mergedRules = append(mergedRules, denyRules.Rules...)
+				slices.Sort(mergedRules)
+				mergedRulesInFilter := RulesInFilter{Filter: allowRules.Filter, Rules: mergedRules}
+				mergedRulesInLayer = append(mergedRulesInLayer, mergedRulesInFilter)
+			}
+		}
+		allowDenyMerged[layer] = mergedRulesInLayer
+	}
+	return allowDenyMerged
+}
+
+// allow and deny in layer: gets all indexes of a layer (e.g. indexes of nacls)
+func getAllIndexesForFilter(allowForLayer, denyForLayer []RulesInFilter) []int {
+	indexesAllow := getIndexesOfFilters(allowForLayer)
+	indexesDeny := getIndexesOfFilters(denyForLayer)
+	allIndexes := indexesAllow
+	for _, indx := range indexesDeny {
+		if !slices.Contains(allIndexes, indx) {
+			allIndexes = append(allIndexes, indx)
+		}
+	}
+	sort.Ints(allIndexes)
```

Comment:
from Ints() documentation:
"Ints sorts a slice of ints in increasing order.

Note: consider using the newer slices.Sort function, which runs faster.
"